src/rdiff-backup-delete

```diff
@@ -60,6 +64,71 @@ def _str(value):
         return value.decode('utf-8', errors='replace')
 
 
+# Check if zcat is available.
+GZIP = spawn.find_executable('gzip')
+
+
+class _wrap_close:
+    """
+    Helper for _open() -- a proxy for a file whose close waits for the process.
+    """
+
+    def __init__(self, stream, proc):
+        self._stream = stream
+        self._proc = proc
+
+    def close(self):
+        if self._proc.stdin:
+            self._proc.stdin.close()
+        returncode = self._proc.wait()
+        if self._proc.stdout:
+            self._proc.stdout.close()
+        if returncode == 0:
+            return None
+        return returncode
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+
+    def __getattr__(self, name):
+        return getattr(self._stream, name)
+
+    def __iter__(self):
+        return iter(self._stream)
+
+
+def _open(fn, mode):
+    """
+    Wrapp to open a file with or without compression using zcat/gzip or
+    pure-python implementation.
+    """
+    compress = fn.endswith(b'.gz')
+    buffered = io.BufferedReader if 'r' in mode else io.BufferedWriter
+    # Open file directly if compression is not required
+    if not compress:
+        return buffered(open(fn, mode))
+
+    # Open file using python gzip if zcat and gzip are not available.
+    if not GZIP:
+        return buffered(gzip.open(fn, mode))
+
+    # When available, open file using subprocess gzip for better performance
+    if 'r' in mode:
+        proc = subprocess.Popen([_bytes(GZIP), b'-cd', fn],
+                                shell=False,
```

Comment:
shell=False is the default and isn't required

Reply:
fixed